internal/api_server/middleware/route_exists.go

```diff
@@ -0,0 +1,22 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+)
+
+func CreateRouteExistsMiddleware(router chi.Router) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			routeContext := chi.NewRouteContext()
+			if !router.Match(routeContext, r.Method, r.URL.Path) {
+				http.NotFound(w, r)
+				return
+			}
+			next.ServeHTTP(w, r)
+		}
+		return http.HandlerFunc(fn)
+	}
+
+}
```

Comment:
I'm ok with that for now, but wondering whether we can avoid the additional search through the routing tree by using route groups to separate what needs an AuthN or AuthZ check, then making the AuthN/AuthZ themselves look up whether a resource exists or not.